python/mxnet/autograd.py

```diff
@@ -112,6 +137,34 @@ def pause(is_train=False):
     return RecordingStateScope(False, is_train)
 
 
+def override_train():
```

Comment:
I agree that we should only introduce features that users need. Just to elaborate on the reasoning: set_training() is currently action-based API, and users will need to manually revert the values if the change is intended for only a small subset, and thus will need to keep the previous states. Introducing train() and predict() scopes can help making this simpler since the "with" scope is a natural fit for such task. @zackchase what's your view as a user?